### PR TITLE
Add basic Cypress tests for waterfall charts

### PR DESCRIPTION
**Pull request by ariya**
How to verify? Run the tests as usual.

```
yarn test-cypress-no-build
```

### CODE REVIEW

File: frontend/test/metabase/scenarios/visualizations/waterfall.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      cy.findByText("Total");
+    });
+  });
+
+  it("should work with time-series data", () => {
+    cy.visit("/");
+    cy.findByText("Ask a question").click();
+    cy.findByText("Custom question").click();
+    cy.findByText("Sample Dataset").click();
+    cy.findByText("Orders").click();
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/visualizations/waterfall.cy.spec.js:65]: `openOrdersTable({ mode: "notebook" })`

Of course, import openOrdersTable before this :)

Comment (ariya) [reply frontend/test/metabase/scenarios/visualizations/waterfall.cy.spec.js]: Thanks! I've adopted this syle.

**Review by nemanjaglumac — APPROVED** [review]
Everything looks good to me. Tested it locally and it works like a charm :)

One question, though: How important it is to use "manual" reproduction (for creating questions, for example) vs using the API to achieve the same thing?

Tests took between 35 and 40s to finish locally. With the use of API in question creation, we could reduce that significantly.

**Comment by ariya** [comment]
> Tests took between 35 and 40s to finish locally. With the use of API in question creation, we could reduce that significantly.

If the tests are not about preparing the data (e.g. the specific series for this waterfall rendering), then surely we can skip the "manual repro" and speed it up by using the API. Perhaps, we need to strike balance. For instance, the first step could be a full-blown "manual repro" (to ensure we don't have a blind spot here) but the subsequent ones can enjoy some shortcuts.

**Comment by nemanjaglumac** [comment]
Would you like me to "jump in" and prepare API for this or you can simply merge it as it is, and I'll do that refactoring later? There will be a concentrated effort to "convert" a lot of our tests to use API, anyway.

**Comment by ariya** [comment]
> Would you like me to "jump in" and prepare API for this or you can simply merge it as it is, and I'll do that refactoring later? There will be a concentrated effort to "convert" a lot of our tests to use API, anyway.

I'll probably merge it soon, just because we now have zero coverage on waterfall E2E tests. But yeah, definitely looking forward to having the conversion happening soon!

**Pull request merged**